p2pserver/link/link.go

```diff
@@ -118,10 +138,25 @@ func (this *Link) Rx() {
 	reader := bufio.NewReaderSize(conn, common.MAX_BUF_LEN)
 
 	for {
+		this.lock.Lock()
+		for this.err == nil && this.writeBuff.Size() > uint64(this.maxWriteBuff) {
+			this.cond.Wait()
+		}
+		if this.err != nil {
+			this.lock.Unlock()
+			break
+		}
+		this.lock.Unlock()
```

Comment:
move this line to Line145?

Reply:
access `this.err` should be in locked context